### DISCUSSION TITLE
Gold Knob submenus for easy access and intuitive parameter editing

### DISCUSSION THREAD

**Discussion by t2fr2** [SynthstromAudible/DelugeFirmware#2525] [Ideas]
Hi, all!

Idk if this only fits my workflow, or if I'm missing something, but I always feel in need of knobs when playing the deluge

as of now, we edit parameters either with predetermined or custom shortcuts, by shift pressing on the grid and the select knob, or by menu diving the select knob. only the predetermined and custom parameters are immediate between idea and execution (one click on a button in a very easy to remember line, and either top or botton knob) but even then, in the case of filters/eq, in the moment I sometimes miss the click page and have to cycle through them again. this makes the machine performance friendly enough daw style, but I feel it lacks as an instrument to play without external controllers or gear, or as a tool for working with improvisation and composing out of grid and out of overlapping sequences of loops and variations.

This workflow barrier is also apparent in sound design, when creating synths by manually pressing oscillators, or editing fx quickly with more than just small, mid, large room size on a reverb, and with the hidden parameters (I think mutable reverb is not fully accessible on the grid).

my idea is a gold knob submenu, similar to how old matrix style multifx guitar pedals work, either by having pages of one or more fx or osc like reverb/delay (8 buttons and 2 knobs are 16 "easy access" parameters), that you could either access through shift+clicking the parameter, and it would open that parameters tree, or a general combo like shif+keyboard to open the 1st submenu. submenus could be scrolled in the select knob like automations do, with the same sequence (grouped) to make it easy to navigate and remember both. 

ex: reverb pan opens the reverb golden knob submenu, with the first goldenknob button (level/pan) already pre-selected - to keep pans in the same place always, pressing the cutoff/res button would give me width and dampening or other parameters, turning the select knob right would take me to the delay, where i could change level, feedback, time and type, also keeps the possibility open for new tweekable parameters. OSC1 page or OSC 1+2 combo (top knob are OSC1 parameters and 2 is bottom, or first 4 + last 4) would make sound designing so much more intuitive and fast. 7SEG could say the name of the page, and parameter while tweaking - value is the led; OLED could have both up and down golden knob functions

this could make designing and composing way more deep and could turn tweakability part of performance either for live performance or arranger recording, like a dedicated fx pedal or knobs in a synth usually do. this can also make the deluge more suitable for other fx in the future

thanks for reading and for all the good work everybody has put in this community firmware. this machine was good, now it's the best. I only know how to print hello world on a piece of paper, but I can help with designing and planning, if anyone wants to pick this up

**Comment by seangoodvibes** [comment]
Idk if this will address your request but someone is already working on an updated menu system which requires less menu diving.

Basically if you press a shortcut on the grid it will take you into a horizontal menu that has all the parameters related to that shortcut side by side.

Eg shift lpf freq will take you into a menu where you have all the lpf parameters side by side

I think this will make sound design much easier

Also, if you haven't tried it as well, another great technique is simply going into automation view and scrolling left and right using the select encoder. You have quick access to all the parameters and they are grouped together. In other words, you don't have to use automation view strictly for automation. It can be a quick way of visualizing parameter values on the grid and using the gold knobs to increase / decrease the parameter values.

It's not impossible to get a menu like system on the gold knobs either. It was done previously with a compressor UI. Just takes someone willing to do the work.

Lastly, if you forget what parameters are currently activated on the gold knobs, you can press and hold the white buttons between the gold knobs to see what is mapped to the gold knobs associated with that button.

![image](https://github.com/user-attachments/assets/c51c6697-ac08-44b4-880b-07ecf5b83171)

![image](https://github.com/user-attachments/assets/90a827b4-6d93-4733-bfdb-8f50b70a187b)

![image](https://github.com/user-attachments/assets/13984edc-f39e-4445-b4df-ad177e9e3c7f)

**Comment by t2fr2** [comment]
thanks for taking the time to reply!

that select knob menu will sure make things a lot easier for designing, and I've been using the automation for quick access, the changing pages is what gave me this idea

I think both lack in the immediacy, a way to work without having to look to know what you're doing. why an ESX's fx section is performative with 2 knobs, and why a zoom MS-70 with 3 knobs isn't

I know dreaming is easy, and work needs to be done, I surely don't know how do it, so I can only write

**Comment by nikodemus** [comment]
I haven't thought about this longer than 10s, but in the abstract I like the idea of "gold knob menus" to access more parameters fastish. Excactly what and exactly how, I don't know... but it's a thought for sure.
